### PR TITLE
feat(lib-inject): support py3.12

### PR DESCRIPTION
**Pull request by mabdinur**
Support single step apm instrumentation for applications using python3.12

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.
- [x] If change touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: lib-injection/dl_wheels.py
Comment (Kyle-Verhoog) [inline lib-injection/dl_wheels.py:42]: note that these are supported versions that the script itself supports. It still supports 2.7 and 3.6 we just don't actually use them when calling the script.

Are you sure you want to do away with them here?

Comment (mabdinur) [reply lib-injection/dl_wheels.py:42]: Leaving this as is won't break anything but it could confuse maintainers. This array implies python<3.7 is supported on the main branch. I also don't see a scenario where we will run this script with changes in 1.x.

Comment (mabdinur) [reply lib-injection/dl_wheels.py:42]: Okay you win, I will add it back

File: lib-injection/dl_wheels.py
Comment (mabdinur) [inline lib-injection/dl_wheels.py:42]:
```suggestion
supported_versions = ["2.7", "3.6", "3.7", "3.8", "3.9", "3.10", "3.11", "3.12"]
```

File: releasenotes/notes/single-step-3-12-support-45b60e50eefac26c.yaml
Comment (mabdinur) [inline releasenotes/notes/single-step-3-12-support-45b60e50eefac26c.yaml:2]:
```suggestion
```

File: lib-injection/dl_wheels.py
Comment (Kyle-Verhoog) [inline lib-injection/dl_wheels.py:80]: revert this change too? You reverted the above

**Review by Kyle-Verhoog — COMMENTED** [review]
just one comment. Good to go after it is resolved.

File: lib-injection/dl_wheels.py
Comment (mabdinur) [inline lib-injection/dl_wheels.py:79]:
```suggestion
        # Have to special-case these versions of Python for some reason.
        if python_version in ["2.7", "3.5", "3.6", "3.7"]:
```

File: lib-injection/dl_wheels.py
Comment (mabdinur) [inline lib-injection/dl_wheels.py:81]:
```suggestion
```

**Review by Kyle-Verhoog — APPROVED** [review]

File: releasenotes/notes/single-step-3-12-support-45b60e50eefac26c.yaml
Comment (mabdinur) [inline releasenotes/notes/single-step-3-12-support-45b60e50eefac26c.yaml:4]:
```suggestion
    lib-injection: Adds supports for installing ddtrace via single step instrumentation on python3.12
```

File: tests/lib-injection/dd-lib-python-init-test-django-uvicorn/Dockerfile
Comment (erikayasuda) [inline tests/lib-injection/dd-lib-python-init-test-django-uvicorn/Dockerfile:1]: Why does the python and django version upgrade only happen for this Dockerfile?

**Review by erikayasuda — COMMENTED** [review]
Just one question about the test changes you made

**Review by erikayasuda — APPROVED** [review]
LGTM

File: tests/lib-injection/dd-lib-python-init-test-django-uvicorn/Dockerfile
Comment (erikayasuda) [reply tests/lib-injection/dd-lib-python-init-test-django-uvicorn/Dockerfile:1]: Answered offline: We needed test coverage for 3.12, and we had two for 3.9, so this was to ensure we had a test case for 3.12 since it would be newly supported.

**Pull request merged**
